src/main/java/gift/service/WishService.java

```diff
@@ -24,36 +25,51 @@ public WishService(WishRepository wishRepository, ProductService productService)
 
     @Transactional
     public void save(MemberRequestDto memberRequestDto, WishRequestDto wishRequestDto){
-        ProductResponseDto productDto = productService.findByName(wishRequestDto.getProductName());
-        Product product = new Product(productDto.getId(), productDto.getName(), productDto.getPrice(), productDto.getImageUrl());
+        ProductResponseDto productResponseDto = productService.findByName(wishRequestDto.getProductName());
 
-        wishRepository.save(new Wish(memberRequestDto.toEntity(),product,wishRequestDto.getQuantity()));
+        Product product = productResponseDto.toEntity();
+        Member member = memberRequestDto.toEntity();
+
+        Wish newWish = new Wish(member,product,wishRequestDto.getQuantity());
+
+        product.addWish(newWish);
+        member.addWish(newWish);
+        wishRepository.save(newWish);
     }
 
     @Transactional(readOnly = true)
-    public List<WishResponseDto> getMemberWishListByMemberId(Long memberId){
+    public List<WishResponseDto> getMemberWishesByMemberId(Long memberId){
         return wishRepository.findByMemberId(memberId)
-                .orElseThrow(() -> new WishNotFoundException(Messages.NOT_FOUND_WISH))
                 .stream()
                 .map(WishResponseDto::from)
                 .toList();
 
     }
 
+    @Transactional(readOnly = true)
+    public List<WishResponseDto> getPagedMemberWishesByMemberId(Long memberId, Pageable pageable){
+        return wishRepository.findByMemberId(memberId,pageable)
+                .stream()
+                .map(WishResponseDto::from)
+                .toList();
+    }
+
     @Transactional
-    public void deleteWishByMemberIdAndWishId(Long memberId, Long id){
-        wishRepository.findByIdAndMemberId(id, memberId)
+    public void deleteWishByMemberIdAndId(Long memberId, Long id){
+        Wish wish = wishRepository.findByIdAndMemberId(id, memberId)
                 .orElseThrow(()-> new WishNotFoundException(Messages.NOT_FOUND_WISH));
+
+        wish.getMember().removeWish(wish);
+        wish.getProduct().removeWish(wish);
         wishRepository.deleteById(id);
     }
 
     @Transactional
-    public void updateQuantityByMemberIdAndWishId(Long memberId, Long id, WishRequestDto request){
+    public void updateQuantityByMemberIdAndId(Long memberId, Long id, WishRequestDto request){
         Wish existingWish = wishRepository.findByIdAndMemberId(id, memberId)
                 .orElseThrow(()-> new WishNotFoundException(Messages.NOT_FOUND_WISH));
 
         Wish updatedWish = new Wish(existingWish.getId(),existingWish.getMember(),existingWish.getProduct(),request.getQuantity());
         wishRepository.save(updatedWish);
```

Comment:
```suggestion
existingWish.updateQuantity(request.getQuantity());
```
위와 같이 한줄의 코드로 해결할 수도 있었을 것 같은데 Wish 객체를 만들고 save한 이유가 있을까요?

Reply:
제가 최근에 JPA를 배워서  엔티티의 속성을 변경하기만 하면, 별도로 save 메서드를 호출하지 않아도 트랜잭션이 끝날 때 변경 사항이 자동으로 데이터베이스에 반영된다는 사실을 오늘 제대로 알았습니다!
아래와 같이 수정하겠습니다!